### PR TITLE
[MISC] Rollback Patroni update

### PR DESCRIPTION
**Pull request by dragomirp**
Rollback to Patroni 3.1.2.

### CODE REVIEW

**Comment by dragomirp** [comment]
I'm still seeing the /config endpoint being out of sync, but that seems to be the case with 14/stable as well. We'll have to take a closer look after the sprint.

File: tests/integration/test_charm.py [hunk 1]
```diff
@@ -193,12 +195,13 @@ async def test_postgresql_parameters_change(ops_test: OpsTest) -> None:
...
-        connection.close()
 
-        # Validate each configuration set by Patroni on PostgreSQL.
-        assert settings["max_prepared_transactions"] == "100"
-        assert settings["shared_buffers"] == "128"
-        assert settings["lc_monetary"] == "en_GB.utf8"
+                # Validate each configuration set by Patroni on PostgreSQL.
+                assert settings["max_prepared_transactions"] == "100"
+                assert settings["shared_buffers"] == "128"
+                assert settings["lc_monetary"] == "en_GB.utf8"
```
Comment (dragomirp) [inline tests/integration/test_charm.py:202]: Check each node's config values.

**Review by taurus-forever — APPROVED** [review]
I suspected to see here only snap revision update, but all other changes looks OK. LGTM.

File: tests/integration/test_charm.py (hunk 1 above)
Comment (marceloneppel) [reply tests/integration/test_charm.py:202]: Thanks for catching that and improving it!

**Review by marceloneppel — APPROVED** [review]
LGTM!

**Pull request merged**